custom/icds_reports/management/commands/fix_past_data.py

```diff
@@ -0,0 +1,74 @@
+import os
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+
+
+from django.core.management.base import BaseCommand
+from corehq.util.argparse_types import date_type
+from django.db import connections, transaction
+from corehq.sql_db.connections import get_icds_ucr_citus_db_alias
+
+
+@transaction.atomic
+def _run_custom_sql_script(command, day=None):
+    db_alias = get_icds_ucr_citus_db_alias()
+    if not db_alias:
+        return
+    with connections[db_alias].cursor() as cursor:
+        cursor.execute(command, [day])
+
+
+class Command(BaseCommand):
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            'month',
+            type=date_type,
+            help='The start date (inclusive). format YYYY-MM-DD'
+        )
+
+    def build_data(self, monthly_date):
+        date = monthly_date['record_date']
+        print(f'\n======= Executing for month {date}======\n')
+        for i in range(1, 8):
+            print(f'==============Executing Script {i} =============')
+            if monthly_date["default"] is False and i > 2:
+                path = os.path.join(os.path.dirname(__file__), 'sql_scripts',
+                                'fix_past_data_part_1_{}.sql'.format(i))
+            else:
+                path = os.path.join(os.path.dirname(__file__), 'sql_scripts',
+                                    'fix_past_data_part_{}.sql'.format(i))
+            end_date = date + relativedelta(months=1)
+            context = {
+                'start_date': date.strftime("%Y-%m-%d"),
+                'end_date': end_date.strftime("%Y-%m-%d")
+            }
+            with open(path, "r", encoding='utf-8') as sql_file:
+                sql_to_execute = sql_file.read()
+                sql_to_execute = sql_to_execute % context
+                print(sql_to_execute)
+                _run_custom_sql_script(sql_to_execute)
+
+    def handle(self, *args, **kwargs):
```

Comment:
this allows you to specify and end date but not a start date, which means it is not resumable. This is likely to take a very long time to run, and we will want to do it a few months at a time, so it should be configured to allow picking both a start and end. Alternatively, many of our commands ask you to specify the months directly, which would also make the logic below way simpler.

Reply:
updated